crates/dojo-lang/src/component.rs

```diff
@@ -34,6 +38,33 @@ pub fn handle_component_struct(db: &dyn SyntaxGroup, struct_ast: ast::ItemStruct
         ]),
     )];
 
+    // let is_indexed = get_indexed_attr_value(&struct_ast, db).unwrap_or(false);
+
+    let is_indexed_fn = if indexed {
+        RewriteNode::interpolate_patched(
+            "
+                #[view]
+                fn is_indexed() -> bool {
+                    bool::True(())
```

Comment:
You can use just one `RewriteNode` here, something like

```rust
RewriteNode::interpolate_patched(
"
code .... $retval$
",
HashMap::from(["$retval$".to_string(), RewriteNode::Text(if indexed { "bool::True" } else { "bool::False" }]);
```

Reply:
nice one!